entity-service-client/src/main/java/org/hypertrace/entity/data/service/client/EdsCacheClient.java

```diff
@@ -224,4 +226,72 @@ public void upsertEnrichedEntities(String tenantId, EnrichedEntities enrichedEnt
   public void upsertRelationships(String tenantId, EntityRelationships relationships) {
     client.upsertRelationships(tenantId, relationships);
   }
+
+  public void updateBasedOnChangeEvent(
+      EntityChangeEventKey entityChangeEventKey, EntityChangeEventValue entityChangeEventValue) {
+    LOG.debug("Entity change event is {}, {} ", entityChangeEventKey, entityChangeEventValue);
+    Entity entity;
```

Comment:
nit - avoid the single assignment shared var. Just inline this in the function calls (e.g. `updateCacheValues(entityChangeEventKey, entityChangeEventValue.getUpdateEvent().getLatestEntity())`